src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -854,29 +868,32 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_ep_t, const uct_ep_params_t *params)
         memset(&attr, 0, sizeof(attr));
         uct_rc_iface_fill_attr(&iface->super, &attr, 0);
         uct_ib_exp_qp_fill_attr(&iface->super.super, &attr);
-        status = uct_rc_mlx5_iface_create_qp(iface, &self->tm_qp, &self->tx.wq, 
-                                             &attr);
+        status = uct_rc_mlx5_iface_create_qp(iface, &self->tm_qp, NULL, &attr);
         if (status != UCS_OK) {
             goto err;
         }
 
-        uct_rc_iface_add_qp(&iface->super, &self->super, self->tm_qp.verbs.qp->qp_num);
+        uct_rc_iface_add_qp(&iface->super, &self->super, self->tm_qp.qp_num);
     }
 
     self->tx.wq.bb_max = ucs_min(self->tx.wq.bb_max, iface->tx.bb_max);
     uct_rc_txqp_available_set(&self->super.txqp, self->tx.wq.bb_max);
     return UCS_OK;
 
 err:
-    uct_ib_destroy_qp(self->tx.wq.super.verbs.qp);
+    uct_ib_mlx5_destroy_qp(&self->tx.wq.super);
     return status;
 }
 
-static void uct_rc_mlx5_ep_clean_qp(uct_rc_mlx5_ep_t *ep, struct ibv_qp *qp)
+static void uct_rc_mlx5_ep_clean_qp(uct_rc_mlx5_ep_t *ep, uct_ib_mlx5_qp_t *qp)
 {
     uct_rc_mlx5_iface_common_t *iface = ucs_derived_of(ep->super.super.super.iface,
                                                        uct_rc_mlx5_iface_common_t);
 
+    if (qp->type == UCT_IB_MLX5_QP_TYPE_DEVX) {
+        return;
```

Comment:
pls comment why we don't clean for devx qp
BTW what happens if there are leftover completions when we destroy the QP? i think mlx5 poll_cq will see them and fail to lookup the rc ep, no?